lib/ddtrace/transport.rb

```diff
@@ -119,6 +137,8 @@ def server_error?(code)
     # endpoint. In both cases, we're going to downgrade the transporter encoder so that
     # it will target a stable API.
     def downgrade?(code)
+      return if @api.fetch(:compatibility_mode)
```

Comment:
Should we test compatibility_mode here or the existence of a fallback? Both are equivalent with our data, so this is just a nitpick but I was worried because I could not see the check on the existence of an available fallback.

Reply:
humn, makes sense. I think I'll get rid of the compatibility_mode altogether and just check for the existence of a fallback!

Reply:
Yes the `compatibility_mode` was added to the Python client because at that time we had a different data model + JSON/Msgpack encoders. Now, nobody uses such old trace-agent. We can just use the fallback.